### PR TITLE
don't process unpublished courses

### PR DESCRIPTION
**Pull request by gumaerc**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Related to https://github.com/mitodl/hugo-course-publisher/issues/436
Closes https://github.com/mitodl/ocw-to-hugo/issues/149

#### What's this PR do?
This PR makes sure a course has not been unpublished before generating markdown and the data template

#### How should this be manually tested?
Run `ocw-to-hugo` with an unpublished course (`8-01sc-classical-mechanics-fall-2016` for example) and ensure that processed markdown and a data template do not appear in your output directory.

### CODE REVIEW

**Comment by pdpinch** [comment]
How/will this address the course that's already in production, described in https://github.com/mitodl/hugo-course-publisher/issues/436 ?

File: src/helpers_test.js [hunk 1]
```diff
@@ -301,3 +311,13 @@ describe("stripS3", () => {
...
+
+describe("isCoursePublished", () => {
+  it("returns true for an published course", () => {
+    assert.isTrue(helpers.isCoursePublished(singleCourseJsonData))
+  })
+
+  it("returns false for an unpublished course", () => {
+    assert.isFalse(helpers.isCoursePublished(unpublishedCourseJsonData))
+  })
+})
```
Comment (mbertrand) [inline src/helpers_test.js:323]: You could test more permutations with something like this:
```javascript
  ;[
    [null, null, false],
    ["", null, false],
    ["2010/03/10 0:0:0.000", null, true],
    ["2010/03/10 0:0:0.000", "", true],
    [null, "2010/03/10 0:0:0.000", false],
    ["2010/03/10 0:0:0.000", "2010/03/11 0:0:0.000", false],
    ["2010/03/11 0:0:0.000", "2010/03/10 0:0:0.000", true]
  ].forEach(([pubDate, unpubDate, published]) => {
    it(`returns ${String(published)} for publish date ${String(pubDate)}, unpublish date ${String(unpubDate)}`, () => {
      const courseData = {
        last_unpublishing_date:       unpubDate,
        last_published_to_production: pubDate
      }
      assert.equal(helpers.isCoursePublished(courseData), published)
    })
  })
  ```

**Review by mbertrand — CHANGES_REQUESTED** [review]

File: src/helpers.js [hunk 2]
```diff
@@ -395,6 +397,23 @@ const stripS3 = text => {
...
+    courseData["last_published_to_production"],
+    INPUT_COURSE_DATE_FORMAT
+  )
+  const lastUnpublishingDate = moment(
+    courseData["last_unpublishing_date"],
+    INPUT_COURSE_DATE_FORMAT
+  )
+  if (
+    !courseData["last_published_to_production"] ||
+    lastUnpublishingDate.isBefore(lastPublishedToProduction)
```
[the remark this answers is not in the document]
Comment (gumaerc) [reply src/helpers.js]: You are correct.  I had the wrong test course for this case, so I was confused.

**Comment by gumaerc** [comment]
@pdpinch What @mbertrand said about this is "The open code would also need to be changed as well to still upload the parsed json if it changes from 'published' to 'unpublished.'"  Basically, we could handle this by deleting the parsed json file from `open-learning-course-data-production`, but I thought it would be good to have logic at this level too to ignore unpublished courses.  That way, if parsed json from an unpublished course gets unintentionally pushed up to the bucket, it' won't be included in a site publish.

**Comment by gumaerc** [comment]
@mbertrand back to you

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**